packages/core/src/authFlows/oidc_auth_code_flow.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Cognite AS
+import { LOCAL_STORAGE_PREFIX } from '../constants';
+import { UserManager, WebStorageStateStore } from 'oidc-client';
+
+export interface OIDCAuthFlowOptions {
+  clientId: string;
+  openIdConfigurationUrl: string;
+  responseMode: string;
+  responseType?: string;
+  extraScope?: string;
+  authenticateParams?: Record<string, any>;
+  refreshParams?: Record<string, any>;
+  loginParams?: Record<string, any>;
+
+  interactionType?: 'REDIRECT' | 'POPUP';
+}
+
+const localStorageKey = `${LOCAL_STORAGE_PREFIX}oidAuthCodeRedirect`;
+
+export class OidcAuthCode {
+  private readonly userManager: UserManager;
+  private readonly options: OIDCAuthFlowOptions;
+
+  constructor(options: OIDCAuthFlowOptions) {
+    this.options = options;
+    const {
+      openIdConfigurationUrl,
+      clientId,
+      responseMode,
+      responseType = 'code',
+      extraScope = '',
+      authenticateParams,
+    } = options;
+    this.userManager = new UserManager({
+      authority: openIdConfigurationUrl,
+      client_id: clientId,
+      response_mode: responseMode,
+      response_type: responseType,
+
+      scope: extraScope,
+      extraQueryParams: authenticateParams,
+      userStore: new WebStorageStateStore({ store: window.localStorage }),
+      loadUserInfo: false,
+
+      redirect_uri: `${window.location.protocol}//${window.location.host}`,
+    });
+  }
+
+  public async init() {
```

Comment:
Maybe remove `public async`

Reply:
I could but its async ;) I can remove public without any side effect, but its good I think to be clear about what methods can be called and what is helper methods. In addition this file is now deleted.